blocks/layout-grid/src/grid/higher-order.js

```diff
@@ -122,9 +122,9 @@ export function withSetPreviewDeviceType() {
 	return withDispatch( ( dispatch ) => {
 		return {
 			setPreviewDeviceType( type ) {
-				const { __experimentalSetPreviewDeviceType } = dispatch(
-					'core/edit-post'
-				);
+				const { __experimentalSetPreviewDeviceType } =
+					dispatch( 'core/edit-site' ) ||
+					dispatch( 'core/edit-post' );
```

Comment:
Just a quick note that this should work for us given the order (both here and in the selects further down in the file).  `core/edit-post` is available in the Site Editor, but `core/edit-site` is not in the post editor.  So as long as that functionality stay the same, this should work as expected.

It's been like this for a while and I don't expect it to change soon, but I wonder if there is a more robust way to determine the editor in use in case it did. 🤔

Reply:
> I wonder if there is a more robust way to determine the editor in use in case it did

That's what me and @jeyip thought, too. We couldn't find anything like that, but maybe there's more that we don't know. Leaving it to WP masters!